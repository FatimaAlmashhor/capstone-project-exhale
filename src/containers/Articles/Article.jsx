import React from 'react';
import FullArticle from '../../components/FullArticle/Article';

const Article = (artId) => {
  return (
    <div className="bg-background-secondary">
      <FullArticle id={artId} />
    </div>
  );
};

export default Article;
