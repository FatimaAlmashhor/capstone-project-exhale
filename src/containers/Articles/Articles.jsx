import React, { useState } from 'react';
import { useTranslation } from 'react-i18next';
import Article from '../../components/ArticleCard';
import { articles } from '../../services/fakeArticleService';
import { ThemeContext } from '../../themContext';

const Articles = () => {
  const { t } = useTranslation();
  const { theme } = React.useContext(ThemeContext);
  const [articlesCollection, setArticles] = useState(articles);
  const [search, setSearch] = useState('');

  const getArticles = () => {
    return articlesCollection.length !== 0 ? (
      <div className="flex flex-wrap mt-10 -mb-4">
        {articlesCollection.map((article) => {
          return (
            <div className="min-w-0 lg:w-1/3 md:w-1/2" key={article.id}>
              <Article
                id={article.id}
                title={article.title}
                img={article.articleImg}
                paragraph={article.text.substring(1, 100)}
                time={article.date}
                articleLink={article.articleLink.replace('///g', '$')}
              />
            </div>
          );
        })}
      </div>
    ) : null;
  };
  const geHeadArticle = () => {
    const headArt = search === '' ? articlesCollection[0] : null;

    return headArt ? (
      <Article
        id={headArt.id}
        title={headArt.title}
        img={headArt.articleImg}
        paragraph={headArt.text.substring(1, 100)}
        time={headArt.date}
        articleLink={headArt.articleLink.replace('///g', '$')}
      />
    ) : (
      <div />
    );
  };
  const filterArticle = (ser) => {
    let feltered = articlesCollection;
    if (ser !== '') {
      feltered = articlesCollection.filter((m) =>
        m.title.toString().toUpperCase().startsWith(ser.toString().toUpperCase())
      );
      setArticles(feltered);
    } else setArticles(articles);
  };
  const handleSearch = (ser) => {
    setSearch(ser.value);
    filterArticle(ser.value);
  };
  return (
    <div className=" bg-background-primary">
      <div className="Header flex-grow  w-full h-64 top-0 py-64 bg-background-pinck">
        <svg
          className="absolute buttom-0 h-64 w-full "
          xmlns="http://www.w3.org/2000/svg"
          viewBox="0 0 100 100"
          preserveAspectRatio="none"
        >
          <polygon
            fill={theme === 'dark' ? '#0d2438' : 'white'}
            points="0,100 100,0 100,100"
          />
        </svg>
      </div>
      <div className="pt-56 pb-12 text-center absolute top-0 right-0 left-0">
        <h1 className="text-4xl font-semibold text-blue-700">
          {t('Mental Health Articles & Advices')}
        </h1>
        <div className="box-content flex justify-center mt-4 mb-8 text-lg leading-relaxed text-gray-600">
          <p className="w-2/4 ">{t('Articles paragraph')}</p>
        </div>
        <div className="box-content flex justify-center m-5">
          <div className="box-content flex justify-center w-4/5 md:w-3/5 bg-white rounded-full">
            <input
              type="text"
              name="search"
              placeholder={t('Search')}
              className="w-full px-3 py-1 text-base leading-8 text-gray-700 transition-colors duration-200 ease-in-out bg-gray-100 border border-gray-300 rounded outline-none focus:border-indigo-500"
              onChange={(e) => handleSearch(e.target)}
            />
          </div>
        </div>
      </div>

      <div className="flex flex-col  px-2 md:px-12 lg:px-32 justify-center  w-full ">
        <div className="flex-grow mt-64 md:mt-20 w-full wx-auto justify-center">
          {geHeadArticle()}
        </div>
        <div className="flex w-full justify-center " />
        {getArticles()}
      </div>
    </div>
  );
};

export default Articles;
