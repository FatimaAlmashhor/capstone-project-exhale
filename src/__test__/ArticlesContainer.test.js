import React from 'react';
import renderer from 'react-test-renderer';
import {BrowserRouter as  Router} from 'react-router-dom'
import Articles from '../containers/Articles';

it('renders correctly', () => {
  const tree = renderer
    .create(<Router><Articles /></Router> )
    .toJSON();
  expect(tree).toMatchSnapshot();
});