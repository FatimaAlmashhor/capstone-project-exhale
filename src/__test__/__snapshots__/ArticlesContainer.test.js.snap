// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="text-center"
  >
    <h1
      className="text-4xl lg:text-5xl font-bold leading-tight"
    >
      Articles
    </h1>
    <div
      className="flex box-content justify-center"
    >
      <p
        className="font-mono w-2/4"
      >
        Extreme emotions, attitudes, and behaviors involving weight and food is a kind of mental health problem. Read more about the causes, symptoms and how to get help.
      </p>
    </div>
    <div
      className="flex box-content justify-center"
    >
      <div
        className="flex box-content justify-center bg-white rounded-full shadow-xl w-2/5"
      >
        <input
          className="rounded-l-full w-full py-4 px-6 text-gray-700 leading-tight focus:outline-none"
          id="search"
          placeholder="Search"
          type="text"
        />
        <div
          className="p-4"
        >
          <button
            className="bg-blue-500 text-white rounded-full p-2 hover:bg-blue-400 focus:outline-none w-12 h-12 flex items-center justify-center"
            type="button"
          >
            icon
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="flex flex-col"
  >
    <div
      className="w-full flex-grow"
    >
      <section
        className="blog "
      >
        <div
          className="container px-5 py-24 mx-auto"
        >
          <div
            className="flex flex-wrap -mx-4 -mt-4 -mb-10 sm:-m-4"
          >
            <div
              className="flex flex-col items-center justify-center p-5 mx-auto mb-6 md:mb-0"
            >
              <div
                className="w-full h-56 bg-gray-300 bg-center bg-cover rounded-lg shadow-md "
                style={
                  Object {
                    "backgroundImage": "url(https://cdn.shopify.com/s/files/1/0100/4430/9570/articles/Coming_Out_About_Mental_Health_On_Social_Media.jpg?v=1567939851",
                  }
                }
              />
              <div
                className="p-5 -mt-10 overflow-hidden bg-white rounded-lg shadow-lg "
              >
                <div
                  className="inline-flex header-content"
                >
                  <div
                    className="flex-1 w-4 h-4 m-1 bg-green-100 rounded-full category-badge m"
                  >
                    <div
                      className="w-2 h-2 m-1 bg-green-500 rounded-full"
                    />
                  </div>
                  <div
                    className="flex-1 text-sm text-gray-400 category-title"
                  >
                    10/11/2020
                  </div>
                </div>
                <div
                  className="text-3xl title-post"
                >
                  <h2
                    className="text-blue-500 "
                  >
                    Coming Out About Mental Health On Social Media
                  </h2>
                </div>
                <div
                  className="text-base text-justify summary-post text-1xl "
                >
                  <p>
                    Susanna Harris was sitting in her lab class for her graduate program at the University of North Carolina at Chapel Hill when she received an email that told her she had failed .
                  </p>
                  <button
<<<<<<< HEAD
                    className="bg-green-400 text-white px-2 mt-4 block rounded p-2 text-sm"
=======
                    className="block p-2 px-2 mt-4 text-sm text-white bg-green-400 rounded"
>>>>>>> 9afdc7a457514aa7e0f2f5ace7effbc446aeb882
                    type="button"
                  >
                    <li>
                      <a
<<<<<<< HEAD
                        href="//undefined"
=======
                        href="//99"
>>>>>>> 9afdc7a457514aa7e0f2f5ace7effbc446aeb882
                        onClick={[Function]}
                      >
                        Coming Out About Mental Health On Social Media
                      </a>
                    </li>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </section>
    </div>
    <div
      className="w-full flex "
    />
    <div
      className="flex flex-wrap -mb-4"
    >
      <div
        className="w-1/3"
<<<<<<< HEAD
=======
      >
        <section
          className="blog "
        >
          <div
            className="container px-5 py-24 mx-auto"
          >
            <div
              className="flex flex-wrap -mx-4 -mt-4 -mb-10 sm:-m-4"
            >
              <div
                className="flex flex-col items-center justify-center p-5 mx-auto mb-6 md:mb-0"
              >
                <div
                  className="w-full h-56 bg-gray-300 bg-center bg-cover rounded-lg shadow-md "
                  style={
                    Object {
                      "backgroundImage": "url(https://media.gq-magazine.co.uk/photos/5d1390269a22c2b6ef9478f3/16:9/w_1920,c_limit/depression-gq-7aug18_istock_b.jpg",
                    }
                  }
                />
                <div
                  className="p-5 -mt-10 overflow-hidden bg-white rounded-lg shadow-lg "
                >
                  <div
                    className="inline-flex header-content"
                  >
                    <div
                      className="flex-1 w-4 h-4 m-1 bg-green-100 rounded-full category-badge m"
                    >
                      <div
                        className="w-2 h-2 m-1 bg-green-500 rounded-full"
                      />
                    </div>
                    <div
                      className="flex-1 text-sm text-gray-400 category-title"
                    >
                      March 12th, 2019
                    </div>
                  </div>
                  <div
                    className="text-3xl title-post"
                  >
                    <h2
                      className="text-blue-500 "
                    >
                      What is depression
                    </h2>
                  </div>
                  <div
                    className="text-base text-justify summary-post text-1xl "
                  >
                    <p>
                      epression is a common mental disorder affecting more than 264 million people worldwide. It is chara
                    </p>
                    <button
                      className="block p-2 px-2 mt-4 text-sm text-white bg-green-400 rounded"
                      type="button"
                    >
                      <li>
                        <a
                          href="//1"
                          onClick={[Function]}
                        >
                          What is depression
                        </a>
                      </li>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </section>
      </div>
      <div
        className="w-1/3"
>>>>>>> 9afdc7a457514aa7e0f2f5ace7effbc446aeb882
      >
        <section
          className="blog "
        >
          <div
            className="container px-5 py-24 mx-auto"
          >
            <div
              className="flex flex-wrap -mx-4 -mt-4 -mb-10 sm:-m-4"
            >
              <div
                className="flex flex-col items-center justify-center p-5 mx-auto mb-6 md:mb-0"
              >
                <div
                  className="w-full h-56 bg-gray-300 bg-center bg-cover rounded-lg shadow-md "
                  style={
                    Object {
                      "backgroundImage": "url(https://media.gq-magazine.co.uk/photos/5d1390269a22c2b6ef9478f3/16:9/w_1920,c_limit/depression-gq-7aug18_istock_b.jpg",
                    }
                  }
                />
                <div
                  className="p-5 -mt-10 overflow-hidden bg-white rounded-lg shadow-lg "
                >
                  <div
                    className="inline-flex header-content"
                  >
                    <div
                      className="flex-1 w-4 h-4 m-1 bg-green-100 rounded-full category-badge m"
                    >
                      <div
                        className="w-2 h-2 m-1 bg-green-500 rounded-full"
                      />
                    </div>
                    <div
                      className="flex-1 text-sm text-gray-400 category-title"
                    >
                      March 12th, 2019
                    </div>
                  </div>
                  <div
                    className="text-3xl title-post"
                  >
                    <h2
                      className="text-blue-500 "
                    >
                      What is depression
                    </h2>
                  </div>
                  <div
                    className="text-base text-justify summary-post text-1xl "
                  >
                    <p>
                      epression is a common mental disorder affecting more than 264 million people worldwide. It is chara
                    </p>
                    <button
<<<<<<< HEAD
                      className="bg-green-400 text-white px-2 mt-4 block rounded p-2 text-sm"
=======
                      className="block p-2 px-2 mt-4 text-sm text-white bg-green-400 rounded"
>>>>>>> 9afdc7a457514aa7e0f2f5ace7effbc446aeb882
                      type="button"
                    >
                      <li>
                        <a
<<<<<<< HEAD
                          href="//1"
=======
                          href="//2"
>>>>>>> 9afdc7a457514aa7e0f2f5ace7effbc446aeb882
                          onClick={[Function]}
                        >
                          What is depression
                        </a>
                      </li>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </section>
      </div>
      <div
        className="w-1/3"
      >
        <section
          className="blog "
        >
          <div
            className="container px-5 py-24 mx-auto"
          >
            <div
              className="flex flex-wrap -mx-4 -mt-4 -mb-10 sm:-m-4"
            >
              <div
                className="flex flex-col items-center justify-center p-5 mx-auto mb-6 md:mb-0"
              >
                <div
                  className="w-full h-56 bg-gray-300 bg-center bg-cover rounded-lg shadow-md "
                  style={
                    Object {
                      "backgroundImage": "url(https://media.gq-magazine.co.uk/photos/5d1390269a22c2b6ef9478f3/16:9/w_1920,c_limit/depression-gq-7aug18_istock_b.jpg",
                    }
                  }
                />
                <div
                  className="p-5 -mt-10 overflow-hidden bg-white rounded-lg shadow-lg "
                >
                  <div
                    className="inline-flex header-content"
                  >
                    <div
                      className="flex-1 w-4 h-4 m-1 bg-green-100 rounded-full category-badge m"
                    >
                      <div
                        className="w-2 h-2 m-1 bg-green-500 rounded-full"
                      />
                    </div>
                    <div
                      className="flex-1 text-sm text-gray-400 category-title"
                    >
                      March 12th, 2019
                    </div>
                  </div>
                  <div
                    className="text-3xl title-post"
                  >
                    <h2
                      className="text-blue-500 "
                    >
                      What is depression
                    </h2>
                  </div>
                  <div
                    className="text-base text-justify summary-post text-1xl "
                  >
                    <p>
                      epression is a common mental disorder affecting more than 264 million people worldwide. It is chara
                    </p>
                    <button
<<<<<<< HEAD
                      className="bg-green-400 text-white px-2 mt-4 block rounded p-2 text-sm"
=======
                      className="block p-2 px-2 mt-4 text-sm text-white bg-green-400 rounded"
>>>>>>> 9afdc7a457514aa7e0f2f5ace7effbc446aeb882
                      type="button"
                    >
                      <li>
                        <a
<<<<<<< HEAD
                          href="//2"
=======
                          href="//3"
>>>>>>> 9afdc7a457514aa7e0f2f5ace7effbc446aeb882
                          onClick={[Function]}
                        >
                          What is depression
                        </a>
                      </li>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </section>
      </div>
      <div
        className="w-1/3"
      >
        <section
          className="blog "
        >
          <div
            className="container px-5 py-24 mx-auto"
          >
            <div
              className="flex flex-wrap -mx-4 -mt-4 -mb-10 sm:-m-4"
            >
              <div
                className="flex flex-col items-center justify-center p-5 mx-auto mb-6 md:mb-0"
              >
                <div
                  className="w-full h-56 bg-gray-300 bg-center bg-cover rounded-lg shadow-md "
                  style={
                    Object {
                      "backgroundImage": "url(https://media.gq-magazine.co.uk/photos/5d1390269a22c2b6ef9478f3/16:9/w_1920,c_limit/depression-gq-7aug18_istock_b.jpg",
                    }
                  }
                />
                <div
                  className="p-5 -mt-10 overflow-hidden bg-white rounded-lg shadow-lg "
                >
                  <div
                    className="inline-flex header-content"
                  >
                    <div
                      className="flex-1 w-4 h-4 m-1 bg-green-100 rounded-full category-badge m"
                    >
                      <div
                        className="w-2 h-2 m-1 bg-green-500 rounded-full"
                      />
                    </div>
                    <div
                      className="flex-1 text-sm text-gray-400 category-title"
                    >
                      March 12th, 2019
                    </div>
                  </div>
                  <div
                    className="text-3xl title-post"
                  >
                    <h2
                      className="text-blue-500 "
                    >
                      What is depression
                    </h2>
                  </div>
                  <div
                    className="text-base text-justify summary-post text-1xl "
                  >
                    <p>
                      epression is a common mental disorder affecting more than 264 million people worldwide. It is chara
                    </p>
                    <button
<<<<<<< HEAD
                      className="bg-green-400 text-white px-2 mt-4 block rounded p-2 text-sm"
=======
                      className="block p-2 px-2 mt-4 text-sm text-white bg-green-400 rounded"
>>>>>>> 9afdc7a457514aa7e0f2f5ace7effbc446aeb882
                      type="button"
                    >
                      <li>
                        <a
<<<<<<< HEAD
                          href="//3"
=======
                          href="//4"
>>>>>>> 9afdc7a457514aa7e0f2f5ace7effbc446aeb882
                          onClick={[Function]}
                        >
                          What is depression
                        </a>
                      </li>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </section>
      </div>
      <div
        className="w-1/3"
      >
        <section
          className="blog "
        >
          <div
            className="container px-5 py-24 mx-auto"
          >
            <div
              className="flex flex-wrap -mx-4 -mt-4 -mb-10 sm:-m-4"
            >
              <div
                className="flex flex-col items-center justify-center p-5 mx-auto mb-6 md:mb-0"
              >
                <div
                  className="w-full h-56 bg-gray-300 bg-center bg-cover rounded-lg shadow-md "
                  style={
                    Object {
                      "backgroundImage": "url(https://media.gq-magazine.co.uk/photos/5d1390269a22c2b6ef9478f3/16:9/w_1920,c_limit/depression-gq-7aug18_istock_b.jpg",
                    }
                  }
                />
                <div
                  className="p-5 -mt-10 overflow-hidden bg-white rounded-lg shadow-lg "
                >
                  <div
                    className="inline-flex header-content"
                  >
                    <div
                      className="flex-1 w-4 h-4 m-1 bg-green-100 rounded-full category-badge m"
                    >
                      <div
                        className="w-2 h-2 m-1 bg-green-500 rounded-full"
                      />
                    </div>
                    <div
                      className="flex-1 text-sm text-gray-400 category-title"
                    >
                      March 12th, 2019
                    </div>
                  </div>
                  <div
                    className="text-3xl title-post"
                  >
                    <h2
                      className="text-blue-500 "
                    >
                      What is depression
                    </h2>
                  </div>
                  <div
                    className="text-base text-justify summary-post text-1xl "
                  >
                    <p>
                      epression is a common mental disorder affecting more than 264 million people worldwide. It is chara
                    </p>
                    <button
<<<<<<< HEAD
                      className="bg-green-400 text-white px-2 mt-4 block rounded p-2 text-sm"
=======
                      className="block p-2 px-2 mt-4 text-sm text-white bg-green-400 rounded"
>>>>>>> 9afdc7a457514aa7e0f2f5ace7effbc446aeb882
                      type="button"
                    >
                      <li>
                        <a
<<<<<<< HEAD
                          href="//4"
                          onClick={[Function]}
                        >
                          What is depression
                        </a>
                      </li>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </section>
      </div>
      <div
        className="w-1/3"
      >
        <section
          className="blog "
        >
          <div
            className="container px-5 py-24 mx-auto"
          >
            <div
              className="flex flex-wrap sm:-m-4 -mx-4 -mb-10 -mt-4"
            >
              <div
                className="p-5  md:mb-0 mb-6 flex flex-col justify-center items-center  mx-auto"
              >
                <div
                  className="bg-gray-300 h-56 w-full rounded-lg shadow-md bg-cover bg-center "
                  style={
                    Object {
                      "backgroundImage": "url(https://media.gq-magazine.co.uk/photos/5d1390269a22c2b6ef9478f3/16:9/w_1920,c_limit/depression-gq-7aug18_istock_b.jpg",
                    }
                  }
                />
                <div
                  className="  bg-white -mt-10 shadow-lg rounded-lg overflow-hidden p-5"
                >
                  <div
                    className="header-content inline-flex"
                  >
                    <div
                      className="category-badge flex-1  h-4 w-4 m rounded-full m-1 bg-green-100"
                    >
                      <div
                        className="h-2 w-2 rounded-full m-1 bg-green-500"
                      />
                    </div>
                    <div
                      className="category-title flex-1 text-sm text-gray-400"
                    >
                      March 12th, 2019
                    </div>
                  </div>
                  <div
                    className="title-post text-3xl"
                  >
                    <h2
                      className="text-blue-500 "
                    >
                      What is depression
                    </h2>
                  </div>
                  <div
                    className="summary-post text-base text-justify text-1xl "
                  >
                    <p>
                      epression is a common mental disorder affecting more than 264 million people worldwide. It is chara
                    </p>
                    <button
                      className="bg-green-400 text-white px-2 mt-4 block rounded p-2 text-sm"
                      type="button"
                    >
                      <li>
                        <a
=======
>>>>>>> 9afdc7a457514aa7e0f2f5ace7effbc446aeb882
                          href="//5"
                          onClick={[Function]}
                        >
                          What is depression
                        </a>
                      </li>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </section>
      </div>
    </div>
  </div>
</div>
`;
